src/main/webapp/definitions/alfresco-search-sql.yaml

```diff
@@ -0,0 +1,316 @@
+  version: '1'
+  title: Alfresco Insight Engine REST API
+basePath: /alfresco/api/-default-/public/search/versions/1/sql
+securityDefinitions:
+  basicAuth:
+    type: basic
+    description: HTTP Basic Authentication
+security:
+  - basicAuth: []
+consumes:
+  - application/json
+produces:
+  - application/json
+paths:
+  '/sql':
+    post:
+      x-alfresco-since: "6.0"
+      tags:
+        - sql
+      summary: Alfresco Insight Engine SQL Passthrough
+      description: |
+        **Note**: this endpoint is available in Alfresco 6.0 and newer versions.
+        This will require Insight Engine and will not work with Alfresco Search Services.
+
+        **You specify all the parameters in this API in a JSON body**,
+        A basic query looks like this:
+
+        ```JSON
+        {
+          "stmt": "select * from alfresco",
+          "locales": ["en_UK"]
+          "timezone": "Europe/London"
+          "includeMetadata":true
+        }
+        ```
+
+        **Note:** the minimum possible query parameter required.
+        ```JSON
+        {
+          "stmt":
+        }
+        ```
+        The expected reponse will appear in the Alfresco format as seen below.
+        ```JSON
+        {
+          "list": {
+            "pagination": {
+              "count": 1,
+              "hasMoreItems": false,
+              "totalItems": 1,
+              "skipCount": 0,
+              "maxItems": 100
+          },
+          "entries": [{
+            "entry": [
+              {
+                "label": "aliases",
+                "value": "{\"SITE\":\"site\"}"
+              },
+              {
+                "label": "isMetadata",
+                "value": "true"
+              },
+              {
+                "label": "fields",
+                "value": "[\"SITE\"]"
+              }
+            ]
+          }]}}
+          ```
+          To override the default format set the format to solr.
+          ```JSON
+          {
+            "stmt": "select * from alfresco",
+            "format": "solr"
+          }
+        ```
+        This will return solr's output response.
+        ```JSON
+        {
+          "result-set": {
+          "docs": [
+            {
+              "aliases": {
+              "SITE": "site"
+            },
+              "isMetadata": true,
+              "fields": [ "SITE"]
+            },
+            {
+                "RESPONSE_TIME": 23,
+                "EOF": true
+            }
+          ]}
+        }
+        ```
+
+
+        You can use the **locales parameter** to filter results based on locale.
+        ```JSON
+        "locales": ["en_UK", "en_US"]
+        ```
+
+        To include timezone in the query add the **timezone parameter**.
+        ```JSON
+        "timezone": "Japan"
+        ```
+
+        You can use the **includeMetadata parameter** to include addtional  information, this is by default set to false.
+
+        ```JSON
+        "includeMetadata": "false"
```

Comment:
Boolean value or String value? I think both will work but we should keep it consistent in the spec.

Reply:
In `SQLSearchRequest` it's specified as boolean